docs/HOW_TO_RUN_TESTS.md

```
@@ -0,0 +1,487 @@
+This guide covers all testing approaches in Otter Empire, from unit tests to headless simulations to E2E tests.
+
+## Quick Reference
+
+```bash
+# Run all tests in monorepo
+pnpm test
+
+# Run tests for specific package
+cd packages/ai-core && pnpm test
+cd packages/ecs && pnpm test
+cd apps/game && pnpm test
```

markdownlint-cli2 (0.18.1)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)